django/db/models/expressions.py

```diff
@@ -960,7 +965,7 @@ class Subquery(Expression):
     def __init__(self, queryset, output_field=None, **extra):
         self.queryset = queryset
         self.extra = extra
-        if output_field is None and len(self.queryset.query.select) == 1:
+        if output_field is None and len(self.queryset.query.select) == 1 and not isinstance(self, Exists):
```

Comment:
I'm not sure what this is guarding against?

Reply:
If it’s not there then `Exists.output_field` could be set to a type that is not `BooleanField`.

Reply:
Indeed, it would be set to the field type of the single selected column (often the PK), in the common case of not explicitly duplicates lying a value in the constructor. It’s necessary because a change a while back made that attribute into a regular (or cached) attribute, instead of a defined @property)

Reply:
Ah yes I see. Exists being a subclass of Subquery.

Reply:
Yeah. I'm not sure if there is a nicer way to write this particular guard.

Reply:
@schinckel this workaround shouldn't necessary once #9765 is merged.